frontend/src/scenes/session-recordings/components/AiRegexHelper/AiRegexHelper.tsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * @fileoverview A component that helps you to generate regex for your settings using Max AI
+ */
+
+import { IconCopy } from '@posthog/icons'
+import { LemonBanner, LemonButton, LemonModal, LemonTextArea } from '@posthog/lemon-ui'
+import { useActions, useValues } from 'kea'
+import { maxGlobalLogic } from 'scenes/max/maxGlobalLogic'
+
+import { AiConsentPopover } from '../AiConsentPopover'
+import { aiRegexHelperLogic } from './aiRegexHelperLogic'
+
+export function AiRegexHelper(): JSX.Element {
+    const filterLogic = aiRegexHelperLogic()
+    const { isOpen, input, generatedRegex, error, isLoading } = useValues(filterLogic)
+    const { setInput, handleGenerateRegex, handleApplyRegex, onClose } = useActions(filterLogic)
+    const { dataProcessingAccepted } = useValues(maxGlobalLogic)
+
+    return (
+        <>
+            <LemonModal isOpen={isOpen} onClose={onClose} title="Max AI Regex Helper">
+                Explain your regex in natural language:
+                <LemonTextArea
+                    placeholder="I want an regex that covers all urls that include 'app.posthog.com/auth/*'"
+                    className="w-full my-2"
+                    maxRows={4}
+                    minRows={2}
+                    value={input}
+                    onChange={(value) => setInput(value)}
+                />
+                <AiConsentPopover />
+                <div className="flex justify-center mt-2">
+                    <LemonButton
+                        type="primary"
+                        onClick={handleGenerateRegex}
+                        disabled={!input.length || isLoading || !dataProcessingAccepted}
+                        loading={isLoading}
+                    >
+                        {generatedRegex ? 'Regenerate' : 'Generate Regex'}
+                    </LemonButton>
+                </div>
+                {generatedRegex && (
+                    <div className="mt-2">
+                        Your regex is:
+                        <div className="flex flex-row justify-between gap-2 items-center">
+                            <LemonBanner
+                                type="info"
+                                className="w-full flex flex-row justify-between gap-2 items-center"
+                            >
+                                {generatedRegex}
```

Comment:
i wondered if this should be a <pre> or <code> block to give it some visual distinction as the result